core/src/avm2/domain.rs

```diff
@@ -121,25 +121,36 @@ impl<'gc> Domain<'gc> {
         Ok(None)
     }
 
+    /// Resolve a Multiname and return the script that provided it.
+    ///
+    /// If a name does not exist or cannot be resolved, an error will be thrown.
+    pub fn find_defining_script(
+        self,
+        activation: &mut Activation<'_, 'gc>,
+        multiname: &Multiname<'gc>,
+    ) -> Result<(QName<'gc>, Script<'gc>), Error<'gc>> {
```

Comment:
just a question: do you think we still need this dance with `get_defining_script` returning a QName that we later use to `get_property`? As in, can't we just use the input multiname directly?
There might be some good motivation as to why we did it this way originally, but it could also be just "get_property used to take a QName arg, so we needed to grab the correct namespace".

Reply:
(this is a note for possible future simplification, don't need to simplify this in this PR :) )